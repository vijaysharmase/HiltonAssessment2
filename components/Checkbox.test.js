import React from "react";
import enzyme from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import Checkbox from "./Checkbox";
enzyme.configure({ adapter: new Adapter() });

describe("Checkbox component", () => {
	var checked = true;
	const props = {
		className: "new-checkbox",
		checked: true,
		onChange: event => {
			checked = event.target.checked;
		}
	};

	it("check checkbox change functionality", () => {
		const wrapper = enzyme.mount(<Checkbox {...props} />);
		wrapper.find('input[type="checkbox"]').simulate("change");
		expect(checked).toBe(true);
	});

	it("Compare snapshots", () => {
		const wrapper = renderer.create(<Checkbox {...props} />);
		const checkbox = wrapper.toJSON();
		expect(checkbox).toMatchSnapshot();
	});
});
