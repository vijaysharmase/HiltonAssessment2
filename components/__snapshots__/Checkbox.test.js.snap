// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component Compare snapshots 1`] = `
<div
  className="new-checkbox sc-bdVaJa jBKccC"
>
  <input
    checked={true}
    className="sc-htpNat fXAzKf"
    onChange={[Function]}
    type="checkbox"
  />
  <div
    checked={true}
    className="sc-bxivhb iHZuqy"
  >
    <svg
      className="sc-bwzfXH ikwlIv"
      viewBox="0 0 24 24"
    >
      <polyline
        points="20 6 9 17 4 12"
      />
    </svg>
  </div>
</div>
`;
